internal/consumer/partitionConsumer.go

```diff
@@ -256,6 +256,19 @@ func (p *partitionConsumer) messageLoop(offsetRange *kafka.OffsetRange) {
 	}
 }
 
+// delayMsg try to postpone the consumption of the msg for topic.Delay duration if the current timestamp
+// is too soon compare to the msg.Timestamp when Topic.Delay is not zero.
+func (p *partitionConsumer) delayMsg(timestamp time.Time) {
+	delay := p.topicPartition.Delay
+	// check non-zero msg delay
+	if delay > 0 {
+		lag := time.Now().Sub(timestamp)
+		p.logger.Debug("delay msg deliver for topic", zap.String("topic", p.topicPartition.Name),
```

Comment:
This log message is a little deceptive. If `delay - lag <= 0`, it will not actually sleep at  all, but it will print "delay msg ..." in logs.

Reply:
True, will make some fix.

Reply:
Can you also add delay amount to the debug message with `zap.Duration("sleepDuration", (lag - delay)` so that we can know how long it sleep for?

`p.logger` already includes topic tag (see line 147 of partitionConsumer.go) so you do not need to include `zap.String("topic", topicPartition.Name)`.

Also nit: "delay msg delivery" instead of "delay msg deliver"